### PR TITLE
In-page nav root margin, threshold customizability, resolves issues with USWDS Site

### PR DESCRIPTION
**Pull request by mitchmoccia**
This important PR includes some additional customization functionality that I found is necessary, especially for an in-page nav/intersection observer implementation that needs to work on any site. The root margin and threshold customizations follow the same paradigm as scroll offset.

Here’s an infographic detailing how I customized In-page nav specifically for the USWDS documentation site

image

### CODE REVIEW

File: packages/usa-in-page-navigation/src/index.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ const CURRENT_CLASS = `${PREFIX}-current`;
 const IN_PAGE_NAV_TITLE_TEXT = "On this page";
 const IN_PAGE_NAV_TITLE_HEADING_LEVEL = "h4";
 const IN_PAGE_NAV_SCROLL_OFFSET = 0;
+const IN_PAGE_NAV_ROOT_MARGIN = "0px";
```
Comment (thisisdano) [inline packages/usa-in-page-navigation/src/index.js:13]: This should be `0px 0px 0px 0px`

File: packages/usa-in-page-navigation/src/index.js [hunk 2]
```diff
@@ -10,6 +10,8 @@ const CURRENT_CLASS = `${PREFIX}-current`;
...
+const IN_PAGE_NAV_THRESHOLD = "0";
```
Comment (thisisdano) [inline packages/usa-in-page-navigation/src/index.js:14]: This should be 1

Comment (mitchmoccia) [reply packages/usa-in-page-navigation/src/index.js]: Done

File: packages/usa-in-page-navigation/src/index.js (hunk 1 above)
Comment (mitchmoccia) [reply packages/usa-in-page-navigation/src/index.js]: Done

File: packages/usa-in-page-navigation/src/usa-in-page-navigation.twig [hunk 3]
```diff
@@ -3,7 +3,13 @@
   {% set list_item_class = 'usa-in-page-nav__item' %}
   {% set current_class = 'usa-current' %}
 
-  <aside class="usa-in-page-nav" data-title-text="On this page" data-title-heading-level="h4" data-scroll-offset="0"></aside>
+  <aside class="usa-in-page-nav" 
+    data-title-text="On this page" 
+    data-title-heading-level="h4" 
+    data-scroll-offset="0" 
+    data-root-margin="0px 0px -90% 0px"
+    data-threshold="0"
```
Comment (thisisdano) [inline packages/usa-in-page-navigation/src/usa-in-page-navigation.twig:11]:
```suggestion
    data-root-margin="0px 0px 0px 0px"
    data-threshold="1"
```

**Review by thisisdano — APPROVED** [review]

**Pull request merged**